src/main/java/de/tum/in/www1/artemis/repository/StudentExamRepository.java

```diff
@@ -215,6 +218,22 @@ SELECT COUNT(se)
             """)
     Optional<StudentExam> findByExamIdAndUserId(@Param("examId") long examId, @Param("userId") long userId);
 
+    @Query("""
+            SELECT DISTINCT se
+            FROM StudentExam se
+            WHERE se.testRun = FALSE
+                AND se.exam.id = :examId
+                AND se.user.id = :userId
+                AND se.id = (
+                    SELECT MAX(se2.id)
+                    FROM StudentExam se2
+                    WHERE se2.testRun = FALSE
+                        AND se2.exam.id = :examId
+                        AND se2.user.id = :userId
+                )
+            """)
+    Optional<StudentExam> findLatestByExamIdAndUserId(@Param("examId") long examId, @Param("userId") long userId);
```

Comment:
Here it queries all studentExams twice. Maybe there is a way to simplify this. Another option would be to just get all studentexams of a user, and than get the latest studentExam with java. (Though this would incur more network load, as all the user's studentExams are sent)

Reply:
good point, thank you. I optimised the query by using `ORDER BY se.id DESC LIMIT 1` instead.